micro-rdk/src/native/conn/mdns.rs

```diff
@@ -68,3 +60,40 @@ impl Mdns for NativeMdns {
         Ok(())
     }
 }
+
+impl Drop for NativeMdns {
+    fn drop(&mut self) {
+        let _ = self.daemon().shutdown();
+    }
+}
+impl Mdns for NativeMdns {
+    fn add_service(
+        &mut self,
+        instance_name: &str,
+        service_type: impl AsRef<str>,
+        proto: impl AsRef<str>,
+        port: u16,
+        txt: &[(&str, &str)],
+    ) -> Result<(), MdnsError> {
+        self.add_service(instance_name, service_type, proto, port, txt)
+    }
+    fn set_hostname(&mut self, hostname: &str) -> Result<(), MdnsError> {
+        self.set_hostname(hostname)
+    }
+}
+
+impl Mdns for &mut NativeMdns {
```

Comment:
Asking for my own understanding: why implement the trait both for the `NativeMdns` type and a mutable ref to it? It isn't a pattern I've seen before.

Reply:
the function  `serve_provisioning_async` takes a `mut mdns: M` as a param where M implements Mdns. But we pass a mutable reference to a `NativeMdns` or `Esp32Mdns` to the function. Therefore the Mdns trait has to be implemented for &mut T.  (note that we have to dereference because the argument if the same as `&mut self: &mut NativeMdns` otherwise we would recurse)
There is another case when trait method takes self as a receiver but you what to implemented for &T only